resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -53,12 +57,15 @@
 
     DnsQueryContext(DnsNameResolver parent,
                     InetSocketAddress nameServerAddr,
-                    DnsQuestion question, Promise<AddressedEnvelope<DnsResponse, InetSocketAddress>> promise) {
-
-        this.parent = parent;
-        this.nameServerAddr = nameServerAddr;
-        this.question = question;
-        this.promise = promise;
+                    DnsQuestion question,
+                    List<DnsRecord> additional,
+                    Promise<AddressedEnvelope<DnsResponse, InetSocketAddress>> promise) {
+
+        this.parent = checkNotNull(parent, "parent");
+        this.nameServerAddr = checkNotNull(nameServerAddr, "nameServerAddr");
+        this.question = checkNotNull(question, "question");
+        this.additional = checkNotNull(additional, "additional");
```

Comment:
Just a guess but seems like the goal is more to clarify the API such that this collection is immutable and cannot be modified after the fact. A side effect of this is preventing concurrent modification exceptions etc... I'm not sure a "deep copy" of all objects is justified.